### PR TITLE
Auto link / connect

### PR DESCRIPTION
**Pull request by petersteneteg**
Better handling of linking in port inspectors. Show autolinks when dragging in processors, disable auto links by pressing alt. Pressing shift while dragging when dragging in processors enables auto connection for inports.

### CODE REVIEW

File: include/inviwo/core/network/autolinker.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#include <inviwo/core/network/processornetwork.h>
+
+namespace inviwo {
+
+class Processor;
+
+/**
+ * \brief Utility class to calculate and apply auto linking
+ */
+class IVW_CORE_API AutoLinker {
```
Comment (ResearchDaniel) [inline include/inviwo/core/network/autolinker.h:42]: Deals with property links between processors?
AutoPropertyLinker in that case?

File: include/inviwo/qt/editor/connectiondraghelper.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#include <QPointF>
+#include <warn/pop>
+
+namespace inviwo {
+
+class NetworkEditor;
+class ConnectionDragGraphicsItem;
+class ProcessorOutportGraphicsItem;
+
+class IVW_QTEDITOR_API ConnectionDragHelper : public QObject {
```
Comment (ResearchDaniel) [inline include/inviwo/qt/editor/connectiondraghelper.h:50]: What does this one do? 
Automatically connects a processor's inports when dragging? Also outports?

**Review by ResearchDaniel — APPROVED** [review]
Sounds like a very neat feature!
Some questions and suggestions that can optionally be done:
1. Possible to hint the possibilities to press alt/shift when dragging for the user since this may not be obvious otherwise?
2. Rename *Link to Property*Link to clarify their purpose (i.e. AutoLink to PropertyAutoLink)

File: src/qt/editor/processordraghelper.cpp [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+    autoLinker_->sortAutoLinkCandidates();
+    for (auto& item : autoLinker_->getAutoLinkCandidates()) {
+        const auto& sourceProperties = item.second;
+        if (!sourceProperties.empty()) {
+            auto sourceProperty = sourceProperties.front();
+            auto sourceProcessor = sourceProperty->getOwner()->getProcessor();
+
+            auto it = autoLinks_.find(sourceProcessor);
+            if (it != autoLinks_.end()) {
+                it->second->setEndPoint(pos + zoom * QPointF{-75.0, 0.0},
```
Comment (martinfalk) [inline src/qt/editor/processordraghelper.cpp:249]: this seems quite arbitrary

**Review by martinfalk — APPROVED** [review]
nice work. This feature will come in handy for sure!

File: src/qt/editor/processordraghelper.cpp (hunk 3 above)
Comment (petersteneteg) [reply src/qt/editor/processordraghelper.cpp:249]: Thats half the processor width :)

File: include/inviwo/core/network/autolinker.h (hunk 1 above)
Comment (petersteneteg) [reply include/inviwo/core/network/autolinker.h:42]: We use propertylink and link interchangeable in many places

File: include/inviwo/qt/editor/connectiondraghelper.h (hunk 2 above)
Comment (petersteneteg) [reply include/inviwo/qt/editor/connectiondraghelper.h:50]: This is just for dragging out regular connections

**Comment by MutterOberin** [comment]
I like it! So at least I got some indirect contribution now.

**Pull request merged**